src/ToolUpdate/UpdateCalculator.php

```diff
@@ -81,14 +91,11 @@ private function calculateTasksToExecute(RepositoryInterface $desired, array $to
                 continue;
             }
             // Installed in another version => upgrade.
-            if (!$this->installed->hasTool($name, $tool->getVersion())) {
+            if ($forceReinstall || !$this->installed->hasTool($name, $tool->getVersion())) {
                 $oldVersion = $this->installed->getTool($name, '*');
 
-                $message = 'Will ' . (
-                    version_compare($oldVersion->getVersion(), $tool->getVersion(), '<')
-                        ? 'upgrade '
-                        : 'downgrade '
-                    ) . $name . ' from version ' . $oldVersion->getVersion() . ' to version ' . $tool->getVersion();
+                $message = 'Will ' . $this->describeUpgradeDirection($oldVersion, $tool) . ' ' . $name
```

Comment:
This will show `Will reinstall foo from version 1.0.0 to version 1.0.0`, this is not that pretty.

Should be moved into method below (see also below comment).